File: app/OnyxPenDemo/src/main/java/com/onyx/android/eink/pen/demo/erase/bean/SplitShapeResult.java
package com.onyx.android.eink.pen.demo.erase.bean;

import com.onyx.android.eink.pen.demo.brush.shape.Shape;

import java.util.ArrayList;
import java.util.List;

public class SplitShapeResult {
    private List<Shape> splitShapes = new ArrayList<>();
    private boolean shapeErased;

    public List<Shape> getSplitShapes() {
        return splitShapes;
    }

    public SplitShapeResult setSplitShapes(List<Shape> splitShapes) {
        this.splitShapes = splitShapes;
        return this;
    }

    public boolean isShapeErased() {
        return shapeErased;
    }

    public SplitShapeResult setShapeErased(boolean shapeErased) {
        this.shapeErased = shapeErased;
        return this;
    }
}

File: app/OnyxPenDemo/src/main/java/com/onyx/android/eink/pen/demo/erase/bean/SplitShapeResult.kt
package com.onyx.android.eink.pen.demo.erase.bean

import com.onyx.android.eink.pen.demo.shape.Shape

class SplitShapeResult {
    private var splitShapes: MutableList<Shape?> = ArrayList()
    private var shapeErased: Boolean = false

    fun getSplitShapes(): MutableList<Shape?> = splitShapes

    fun setSplitShapes(splitShapes: MutableList<Shape?>?): SplitShapeResult {
        this.splitShapes = splitShapes ?: ArrayList()
        return this
    }

    fun isShapeErased(): Boolean = shapeErased

    fun setShapeErased(shapeErased: Boolean): SplitShapeResult {
        this.shapeErased = shapeErased
        return this
    }
}
